core/src/main/scala/spark/metrics/MetricsConfig.scala

```
@@ -0,0 +1,75 @@
+package spark.metrics
+
+import java.util.Properties
+import java.io.{File, FileInputStream}
+
+import scala.collection.mutable
+import scala.util.matching.Regex
+
+import spark.Logging
+
+private[spark] class MetricsConfig(val configFile: Option[String]) extends Logging {
+  initLogging()
+
+  val properties = new Properties()
+  val DEFAULT_PREFIX = "*"
+  val INSTANCE_REGEX = "^(\\*|[a-zA-Z]+)\\.(.+)".r
+  var propertyCategories: mutable.HashMap[String, Properties] = null
+
+  private def setDefaultProperties(prop: Properties) {
+    // empty function, any default property can be set here
+  }
+
+  def initilize() {
+    //Add default properties in case there's no properties file
+    setDefaultProperties(properties)
+
+    configFile map { f =>
+      val confFile = new File(f)
+      if (confFile.exists()) {
+        var fis: FileInputStream = null
+        try {
+          fis = new FileInputStream(confFile)
+          properties.load(fis)
+        } finally {
+          fis.close()
+        }
+      } else {
+        logError("Metrics configure file " + f + " is not found, using default metrics")
```

Here we should throw an exception instead of using the default metrics -- we want the user to notice this